hermes/src/price_feeds_metadata.rs

```diff
@@ -0,0 +1,83 @@
+use {
+    crate::{
+        api::types::{
+            AssetType,
+            PriceFeedMetadata,
+        },
+        network::pythnet::fetch_and_store_price_feeds_metadata,
+        state::{
+            cache::AggregateCache,
+            retrieve_price_feeds_metadata,
+        },
+    },
+    anyhow::Result,
+};
+
+pub async fn get_price_feeds_metadata<S>(
+    state: &S,
+    query: Option<String>,
+    asset_type: Option<AssetType>,
+) -> Result<Vec<PriceFeedMetadata>>
+where
+    S: AggregateCache,
+    S: PriceFeedMetadataProvider,
+{
+    match PriceFeedMetadataProvider::get_price_feeds_metadata(state, query, asset_type).await {
+        Ok(price_feeds_with_update_data) => Ok(price_feeds_with_update_data),
+        Err(e) => Err(e),
+    }
+}
+
+
+#[async_trait::async_trait]
+pub trait PriceFeedMetadataProvider {
+    async fn get_price_feeds_metadata(
+        &self,
+        query: Option<String>,
+        asset_type: Option<AssetType>,
+    ) -> Result<Vec<PriceFeedMetadata>>;
+}
+
+#[async_trait::async_trait]
+impl PriceFeedMetadataProvider for crate::state::State {
```

Comment:
I see why you've added this, but the problem with this design is essentially it treats `State` as a type that implements business logic. What will happen if we follow this pattern is `State` will slowly accumulate unrelated logic into a glob of random logic. It's better to have this logic be in some other modular pattern that state is passed into and used, but not _on_ `State`.